### PR TITLE
feat(ngAnimate): introduce `$animate.flush` for unit testing

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: test/ngAnimate/animateSpec.js [hunk 1]
```diff
@@ -5059,8 +5042,8 @@ describe("ngAnimate", function() {
       });
       $rootScope.$digest();
 
-      $animate.triggerReflow();
-      $animate.triggerCallbackPromise();
+      $animate.flush();
+      $animate.flush();
```
Comment (IgorMinar) [inline test/ngAnimate/animateSpec.js]: is this necessary? it should be sufficient to call flush just once.

File: test/ngAnimate/animateSpec.js [hunk 2]
```diff
@@ -5089,12 +5072,12 @@ describe("ngAnimate", function() {
...
```
Comment (IgorMinar) [inline test/ngAnimate/animateSpec.js]: same here

File: src/ngMock/angular-mocks.js [hunk 3]
```diff
@@ -803,6 +803,29 @@ angular.mock.animate = angular.module('ngAnimateMock', ['ng'])
...
+            $$rAF.flush();
+          }
+          if ($$asyncCallback.queue.length) {
+            somethingFlushed = true;
+            this.triggerCallbackEvents();
+          }
+          if ($browser.deferredFns.length) {
+            somethingFlushed = true;
+            this.triggerCallbackPromise();
+          }
```
Comment (IgorMinar) [inline src/ngMock/angular-mocks.js]: we should put this into a loop and keep on flushing stuff until everything settles

**Comment by IgorMinar** [comment]
the rest looks ok, but we need to test this at google to see if there will be issue with flushing browser.deferreds in this way

**Comment by matsko** [comment]
Merged as https://github.com/angular/angular.js/commit/f98e038418f7367b2373adcf4887f64a8e8bdcb0

**Pull request closed without merging**